apps/judicial-system/backend/src/app/modules/defendant/defendant.service.ts

```diff
@@ -203,6 +203,22 @@ export class DefendantService {
   ): Promise<Defendant> {
     const formattedNationalId = formatNationalId(defendantNationalId)
 
+    const defendant = await this.defendantModel.findOne({
+      where: {
+        caseId,
+        [Op.or]: [
+          { nationalId: formattedNationalId },
+          { nationalId: defendantNationalId },
+        ],
+      },
+    })
+
+    if (!defendant) {
+      throw new InternalServerErrorException(
```

Comment:
This is ok, but note that this case is already handled below in the call to `getUpdatedDefendant`. Therefore, consider removing the lookup and exception here.

One thing to be aware of is that national ids are not strictly unique within a case, so in practice there may be multiple defendants with the same national id. However, we do not expect users to do such stupid things on purpose. If multiple defendants have the same national id, then all will be updated below and an error logged. Execution then proceeds as normal.